### ISSUE TITLE
Sass variables removed in version 4?

### ISSUE THREAD

**Issue by benwiley4000** [twbs/bootstrap#19877]
Without knowing a lot about Bootstrap (I'm diving straight into version 4), I'm working with a theme that relies on Bootstrap 3 - I'm trying to put Bootstrap 4 in instead. This particular module wants to use a few variables that aren't around anymore -- [$line-height-computed](https://github.com/twbs/bootstrap/blob/10606a73bcbfc6c723d60e2a1069c921afff0d2a/less/variables.less#L65), [$screen-sm-min](https://github.com/twbs/bootstrap/blob/10606a73bcbfc6c723d60e2a1069c921afff0d2a/less/variables.less#L295), [$navbar-height](https://github.com/twbs/bootstrap/blob/10606a73bcbfc6c723d60e2a1069c921afff0d2a/less/variables.less#L360).

Could anyone explain why those are no longer required by Bootstrap itself and whether I should compute these variables myself or use a replacement variable from the new styles?

**Comment by cvrebert** [comment]
* Regarding ` $screen-sm-min`, see:
https://github.com/twbs/bootstrap/blob/v4-dev/docs/migration.md#responsive-utilities
* The navbar is still in the process of being completely overhauled in v4; you might as well start from scratch (and may have to rewrite it again later; v4 is still in alpha).
* Re ` $line-height-computed`, I'd have to read through the git history...

**Comment by niftylettuce** [comment]
Any update on this?

**Comment by niftylettuce** [comment]
It looks like `$line-height-computed` was removed in v4.  @mdo @cvrebert do we need to add this back?

V3 at <https://github.com/twbs/bootstrap/blob/v3.3.7/less/variables.less#L62-L65>:

```less
//** Unit-less `line-height` for use in components like buttons.
@line-height-base:        1.428571429; // 20/14
//** Computed "line-height" (`font-size` * `line-height`) for use with `margin`, `padding`, etc.
@line-height-computed:    floor((@font-size-base * @line-height-base)); // ~20px
```

V4 at <https://github.com/twbs/bootstrap/blob/v4.0.0-alpha.3/scss/_variables.scss#L174>:

```scss
$line-height-base: 1.5 !default;
```

**Comment by cvrebert** [comment]
The most helpful thing to do would be to look thru the git history and find which commit removed the variable.

**Comment by bardiharborow** [comment]
@cvrebert, I suspect you're looking for ed89be1. (`git bisect` is amazing.)

**Comment by mdo** [comment]
I removed the computed line-height as I don't really need it anymore for setting vertical spacing between elements (e.g., at the bottom of an alert or between headings). Those things are mostly better left to the margin utilities, type scale, etc.

All the screen sizes and what not were redone in v4, so no surprise the `$screen-` vars are gone.

And I've removed, wherever possible, fixed-height values and variables. They are flimsy for creating complex components, are often the result of complex math, etc.

Sorry for the super delayed response!
